### PR TITLE
18788 Updated amalgamation -> amalgamationApplication + business name options (WIP)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#18788

*Description of changes:*
    - app version = 5.6.14
    - added ResultingBusinessName component
    - misc refactor and new code to handle 3 options
    - changed snackbar close text to a white icon
    - added random key to BusinessLookup to reset its state between calls
    - replaced local EntityState with shared EntityStates
    - added validation for isHistorical
    - updated imported enums, etc
    - **updated some enums locally (amalgamation -> amalgamationApplication)**
    - ResultingBusinessName is still WIP (so don't use it)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Amalgamation/AmalgamatingBusinesses.vue [hunk 1]
```diff
@@ -49,8 +49,9 @@
               to add to this application.</span>
 
             <BusinessLookup
+              :key="Math.random()"
```
Comment (severinbeauvais) [inline src/components/Amalgamation/AmalgamatingBusinesses.vue:52]: @JazzarKarim This resets the instance of this component each time it's used, which resets its state, which allows a used to search anew if there was a problem with the previous search.

**Comment by severinbeauvais** [comment]
I think we should merge this PR asap so we can all have the latest change from amalgamation -> amalgamationApplication, even though Resulting Business Name is imcomplete. Thoughts?

File: package.json [hunk 2]
```diff
@@ -23,19 +23,19 @@
     "@bcrs-shared-components/confirm-dialog": "1.2.1",
     "@bcrs-shared-components/contact-info": "1.2.15",
     "@bcrs-shared-components/corp-type-module": "1.0.11",
-    "@bcrs-shared-components/correct-name": "1.0.9",
+    "@bcrs-shared-components/correct-name": "1.0.33",
```
Comment (severinbeauvais) [inline package.json:26]: Since I'm using this shared component, I figured I'd better upgrade to the latest version first.

File: src/App.vue [hunk 3]
```diff
@@ -258,7 +258,7 @@ import { AmalgamationRegResources, DissolutionResources, IncorporationResources,
 import { AuthServices, LegalServices, PayServices } from '@/services/'
 
 // Enums and Constants
-import { EntityState, ErrorTypes, FilingCodes, FilingNames, FilingStatus, FilingTypes, NameRequestStates, RouteNames,
+import { EntityStates, ErrorTypes, FilingCodes, FilingNames, FilingStatus, FilingTypes, NameRequestStates, RouteNames,
```
Comment (severinbeauvais) [inline src/App.vue:261]: I deleted the local enum `EntityState` and am now using the shared one (with the same keys).

**Comment by JazzarKarim** [comment]
> I think we should merge this PR asap so we can all have the latest change from amalgamation -> amalgamationApplication, even though Resulting Business Name is imcomplete. Thoughts?

Agreed.

File: src/components/Amalgamation/AmalgamatingBusinesses.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Amalgamation/AmalgamatingBusinesses.vue:52]: Karim, this is still not perfect, as it seems to refresh too often, but I don't want to spend any more time on it atm (future fix if needed).

File: src/components/Amalgamation/AmalgamatingBusinesses.vue [hunk 4]
```diff
@@ -49,8 +49,9 @@
...
               :showErrors="false"
-              :businessLookup="initialBusinessLookupObject"
+              :businessLookup="{ ...EmptyBusinessLookup }"
```
Comment (severinbeauvais) [inline src/components/Amalgamation/AmalgamatingBusinesses.vue:54]: A simplification.

File: src/components/Amalgamation/AmalgamatingBusinesses.vue (hunk 1 above)
Comment (JazzarKarim) [reply src/components/Amalgamation/AmalgamatingBusinesses.vue:52]: Sounds good to me.

File: src/components/Amalgamation/ResultingBusinessName.vue
Comment (severinbeauvais) [inline src/components/Amalgamation/ResultingBusinessName.vue:1]: This component is WIP, although it partly works atm.

File: src/mixins/amalgamation-mixin.ts [hunk 5]
```diff
@@ -44,6 +45,17 @@ export default class AmalgamationMixin extends Vue {
     return null
   }
 
+  /**
+   * Disallow historical business.
+   * (Could happen if it was added while active and is now historical.)
```
Comment (JazzarKarim) [inline src/mixins/amalgamation-mixin.ts:50]: Great catch!

File: src/components/Amalgamation/ResultingBusinessName.vue [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    return (this.getNameRequestApprovedName || this.getBusinessLegalName)
+  }
+  /** This section's validity state (when prompted by app). */
+  get invalidSection (): boolean {
+    return (this.getShowErrors && !this.getCorrectNameOption)
+  }
+
+  /** Called when component is created. */
+  created (): void {
+    // this.businessNameOption = this.getNameRequestNumber ? 'named' : 'numbered'
```
Comment (JazzarKarim) [inline src/components/Amalgamation/ResultingBusinessName.vue:122]: Eventually, we're going to have to comment this line out right Sev?

Comment (severinbeauvais) [reply src/components/Amalgamation/ResultingBusinessName.vue:122]: Sort of. There will be a new filing variable to save/restore which option was selected (null initially in some cases). I'm not there yet :)

Comment (JazzarKarim) [reply src/components/Amalgamation/ResultingBusinessName.vue:122]: Oops, sorry. All right Sev!

**Pull request merged**